### PR TITLE
Infra primers for torch.export CFR inference preproc (#189424)

### PR DESCRIPTION
**Pull request by bearzx**
Summary:

Infra "primers" that make CFR (main-feed MTML) models runnable through the PyTorch 2 `torch.export` -> Sigmoid/NativeRT path. Each change closes a specific gap hit while exporting, serializing, and serving these nets; together they let a CFR model round-trip through export and execute in the interpreter/NativeRT runtime.

**torch.export serialization** (`caffe2/torch/_export/serde/serialize.py`): serialize `List[List[int]]` and `List[List[float]]` op arguments as `as_int_lists` / `as_float_lists`, including empty nested lists (the element kind is inferred from the schema arg type), and accept arguments passed as actual Python lists rather than only tuples.

**NativeRT `mul.Scalar`** (`caffe2/torch/nativert/kernels/KernelRegistry.cpp`): add `kBool` to the `aten.mul.Scalar` static-kernel dispatch so exported graphs that lift a bool mask into the int/float range (e.g. `aten.gt(...) * 38`) dispatch cleanly, following upstream `aten::mul` type promotion.

**sparsenn torch.export compatibility** (`caffe2/torch/fb/sparsenn/`): add fake/meta implementations for `fb::datafm_sim_hard_search` (plus `_jagged` / `_mt`), `fb::grouped_feature_bitwise_operation`, and `preproc::popcount` so these custom ops trace under export; fix the `ID_LIST` fake to emit 1D int64 offsets matching the runtime op; allocate the `grouped_feature_bitwise_operation` NativeRT output when it is unset; and fix `sigrid_transforms_ops` contiguity (build the ranges tensor with an explicit `c10::kStrided` layout and call `.contiguous()` on the copied tensors) so `caffe2::Tensor`'s `enforce_invariants()` accepts tensors produced by export-traced ops.

**Sigmoid export config** (`sigmoid/inference/export_utils.py`): for PT2 inference export, disable 0/1 specialization (`specialize_int=False`) and enable `automatic_dynamic_shapes` so id-list lengths/values with a sample size of 0 or 1 stay dynamic.

**Localnet benchmark metadata** (`sigrid/feed/prediction_replayer/LocalnetBenchmarkMain.cpp`, `sigrid/predictor/client/localnet/SigridPredictorPytorchLocalModel.cpp`): when `getPredictor()->get_model_metadata()` throws, fall back to `getFeatureMetadataContainer().get_model_metadata("forward")`, which is how exported/sigmoid local nets expose their input metadata.

Unit tests are added for the two open-source changes (serde nested lists and the NativeRT bool `mul.Scalar` dispatch); see the test plan.

Test Plan:
New unit tests covering the two open-source (pytorch) changes:

```
buck2 test fbcode//caffe2/test:test_export -- test_list_of_int_and_float_lists_custom_op
buck2 test fbcode//caffe2/test/cpp/nativert:static_kernel_ops_tests -- StaticKernelTest.MulScalarBool
```

- `test_list_of_int_and_float_lists_custom_op` — defines a custom op with `int[][]` / `float[][]` arguments, exports a module that calls it, and round-trips it through serialize -> deserialize, asserting the nested lists serialize as `as_int_lists` / `as_float_lists` for both non-empty and empty inputs. Result: Pass 1.
- `StaticKernelTest.MulScalarBool` — runs `aten.mul.Scalar` on a bool tensor times int/float scalars through the static kernel and compares against the non-static ATen reference. Result: Pass 1 (full `static_kernel_ops_tests` target: 25/25).

Build verification of the affected C++ targets in dev and opt:

```
buck2 build mode/dev fbcode//caffe2/torch/fb/sparsenn:sparsenn_operators fbcode//sigrid/feed/prediction_replayer:localnet_benchmark fbcode//caffe2/torch/fb/model_transform/fx2trt/packaging:load_net_predictor_lib
buck2 build mode/opt fbcode//sigrid/feed/prediction_replayer:localnet_benchmark fbcode//caffe2/torch/fb/model_transform/fx2trt/packaging:load_net_predictor_lib
```

All builds succeed.

Reviewed By: georgiaphillips, jijiew

Differential Revision: D111165094

### CODE REVIEW

**Comment by angelayi** [comment]
@claude review

**Comment by bearzx** [comment]
Hi @angelayi , this is to extend torch.export so that we can cover a usecase to export preproc module for CFR model.

**Review by angelayi — APPROVED** [review]
